google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcResultSetMetaData.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.jdbc;
+
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+import com.google.cloud.spanner.ResultSet;
+import com.google.common.base.Preconditions;
+
+/** Implementation of {@link ResultSetMetaData} for Cloud Spanner */
+class JdbcResultSetMetaData extends AbstractJdbcWrapper implements ResultSetMetaData {
+
+  private final ResultSet spannerResultSet;
+  private final Statement statement;
+
+  JdbcResultSetMetaData(JdbcResultSet jdbcResultSet, Statement statement) {
+    Preconditions.checkNotNull(jdbcResultSet);
+    this.spannerResultSet = jdbcResultSet.spanner;
+    this.statement = statement;
+  }
+
+  @Override
+  public boolean isClosed() {
+    return false;
+  }
+
+  @Override
+  public int getColumnCount() throws SQLException {
+    return spannerResultSet.getColumnCount();
+  }
+
+  @Override
+  public boolean isAutoIncrement(int column) throws SQLException {
+    return false;
+  }
+
+  @Override
+  public boolean isCaseSensitive(int column) throws SQLException {
+    int type = getColumnType(column);
+    return type == Types.NVARCHAR || type == Types.BINARY;
+  }
+
+  @Override
+  public boolean isSearchable(int column) throws SQLException {
+    return true;
+  }
+
+  @Override
+  public boolean isCurrency(int column) throws SQLException {
+    return false;
+  }
+
+  @Override
+  public int isNullable(int column) throws SQLException {
+    return columnNullableUnknown;
+  }
+
+  @Override
+  public boolean isSigned(int column) throws SQLException {
+    int type = getColumnType(column);
+    return type == Types.DOUBLE || type == Types.BIGINT;
+  }
+
+  @Override
+  public int getColumnDisplaySize(int column) throws SQLException {
+    int colType = getColumnType(column);
+    switch (colType) {
+      case Types.ARRAY:
+        return 50;
+      case Types.BOOLEAN:
+        return 5;
+      case Types.BINARY:
+        int binaryLength = getPrecision(column);
+        return binaryLength == 0 ? 50 : binaryLength;
```

Comment:
let's extract 50 out to a variable - not clear to me what it represents

Reply:
Extracted to a variable and added javadoc to explain its purpose.